Tests/External/Plugins/CodeRefactor.Tests/Commands/RefactorCommandTests.cs

```diff
@@ -350,5 +355,83 @@ internal static string Common(ScintillaControl sci, string sourceText, string fi
                 return sci.Text;
             }
         }
+
+        [TestFixture]
+        public class RenameTests : RefactorCommandTests
+        {
+            static string ReadAllTextAS3(string fileName) => TestFile.ReadAllText(GetFullPathAS3(fileName));
+
+            static string GetFullPathAS3(string fileName) => $"{nameof(CodeRefactor)}.Test_Files.coderefactor.rename.as3.{fileName}.as";
+
+            public IEnumerable<TestCaseData> AS3TestCases
+            {
+                get
+                {
+                    yield return
+                        new TestCaseData("BeforeRenameLocalVariable", 126, "newName")
+                            .Returns(ReadAllTextAS3("AfterRenameLocalVariable"))
+                            .SetName("Rename local variable");
+                }
+            }
+
+            [Test, TestCaseSource(nameof(AS3TestCases))]
+            public string AS3(string fileName, int currentPos, string newName)
+            {
+                ASContext.Context.SetAs3Features();
+                Sci.ConfigurationLanguage = "as3";
+                var sourceText = ReadAllTextAS3(fileName);
+                fileName = GetFullPathAS3(fileName);
+                fileName = Path.GetFileNameWithoutExtension(fileName).Replace('.', Path.DirectorySeparatorChar) + Path.GetExtension(fileName);
+                fileName = Path.GetFullPath(fileName);
+                fileName = fileName.Replace($"\\FlashDevelop\\Bin\\Debug\\{nameof(CodeRefactor)}\\Test_Files\\", $"\\Tests\\External\\Plugins\\{nameof(CodeRefactor)}.Tests\\Test Files\\");
+                ASContext.Context.CurrentModel.FileName = fileName;
+                PluginBase.MainForm.CurrentDocument.FileName.Returns(fileName);
+                return Common(Sci, sourceText, currentPos, newName);
+            }
+
+            static string Common(ScintillaControl sci, string sourceText, int currentPos, string newName)
+            {
+                SetSrc(sci, sourceText, currentPos);
+                var waitHandle = new AutoResetEvent(false);
+                CommandFactoryProvider.GetFactory(sci)
+                        .CreateRenameCommandAndExecute(RefactoringHelper.GetDefaultRefactorTarget(), false, newName)
+                        .OnRefactorComplete += (sender, args) => waitHandle.Set();
+                const int waitTime = 2;
+                var end = DateTime.Now.AddSeconds(waitTime);
+                var result = false;
+                while ((!result) && (DateTime.Now < end))
+                {
+                    Application.DoEvents();
+                    result = waitHandle.WaitOne(0);
+                }
```

Comment:
A bit hacky and would be better to slightly improve some parts of logic, but hacky tests are way better than no tests at all.

Reply:
`Application.DoEvents()` doesn't work for console type applications.

Reply:
I've marked the test to be ignored in the meantime to be able to get builds.